.gitignore

```diff
@@ -10,3 +10,5 @@ tmp
 .classpath
 .project
 .settings
+*.swp
+tags
```

Comment:
If we want to go the route: "`.gitignore` should only be for project-level ignores" I think we'd be left with just:
```
project/project
project/target
target
core/target
tmp
```
What I personally usually do is throw everything in .gitignore, including user-specific files (like files from various editors, operating systems, etc) because we know for sure that we don't want to be checking in .swp files or .classpath files, for example. Whereas, it's possible that someone somewhere might want to make a repo where they check in their .swp files, for some reason or other. So it would be awkward for them switching between projects if we force them to put .swp in `~/.config/git/ignore` just so that can have a clean `git status` when working on this project. But maybe that's not a horrible thing to ask someone to do...?

So that's an argument for allowing all files in here that any contributor wants to ignore. We can agree that we don't want any of these files checked in, and so from that point of view, they are "project-specific" because we don't want them in the project :)